itest/rust/src/virtual_methods_test.rs

```diff
@@ -93,22 +103,91 @@ impl Node2DVirtual for TreeVirtualTest {
 }
 
 #[derive(GodotClass, Debug)]
-#[class(base=Node)]
+#[class(init, base=PrimitiveMesh)]
 struct ReturnVirtualTest {
     #[base]
-    base: Base<Node>,
+    base: Base<PrimitiveMesh>,
 }
 
 #[godot_api]
-impl NodeVirtual for ReturnVirtualTest {
-    fn init(base: Base<Node>) -> Self {
-        ReturnVirtualTest { base }
+impl PrimitiveMeshVirtual for ReturnVirtualTest {
+    fn create_mesh_array(&self) -> VariantArray {
+        varray![
+            PackedVector3Array::from_iter([Vector3::LEFT].into_iter()),
+            PackedVector3Array::from_iter([Vector3::LEFT].into_iter()),
+            PackedFloat32Array::from_iter([0.0, 0.0, 0.0, 1.0].into_iter()),
+            PackedColorArray::from_iter([Color::from_rgb(1.0, 1.0, 1.0)]),
+            PackedVector2Array::from_iter([Vector2::LEFT]),
+            PackedVector2Array::from_iter([Vector2::LEFT]),
+            PackedByteArray::from_iter([0, 1, 2, 3].into_iter()),
+            PackedByteArray::from_iter([0, 1, 2, 3].into_iter()),
+            PackedByteArray::from_iter([0, 1, 2, 3].into_iter()),
+            PackedByteArray::from_iter([0, 1, 2, 3].into_iter()),
+            PackedInt32Array::from_iter([0, 1, 2, 3].into_iter()),
+            PackedFloat32Array::from_iter([0.0, 1.0, 2.0, 3.0].into_iter()),
+            PackedInt32Array::from_iter([0].into_iter()),
+        ]
+    }
+}
+
+#[derive(GodotClass, Debug)]
+#[class(base=Node2D)]
+struct InputVirtualTest {
+    #[base]
+    base: Base<Node2D>,
+    event: Option<Gd<InputEvent>>,
+}
+
+#[godot_api]
+impl Node2DVirtual for InputVirtualTest {
+    fn init(base: Base<Node2D>) -> Self {
+        InputVirtualTest { base, event: None }
+    }
+
+    fn input(&mut self, event: Gd<InputEvent>) {
+        self.event = Some(event);
+    }
+}
+
+#[derive(GodotClass, Debug)]
+#[class(init, base=ResourceFormatLoader)]
+struct FormatLoaderTest {
+    #[base]
+    base: Base<ResourceFormatLoader>,
+}
+
+impl FormatLoaderTest {
+    fn resource_type() -> GodotString {
+        GodotString::from("foo")
+    }
+}
+
+#[godot_api]
+impl ResourceFormatLoaderVirtual for FormatLoaderTest {
+    fn get_recognized_extensions(&self) -> PackedStringArray {
+        [GodotString::from("extension")].into_iter().collect()
+    }
+
+    fn handles_type(&self, type_: godot::prelude::StringName) -> bool {
```

Comment:
Any particular reason why `StringName` is not imported alongside the other builtins?

Reply:
not really, i just had rust analyzer add all default impls and it apparently added it as a fully qualified name and i didnt notice.